servicetalk-examples/http/redirects/src/main/java/io/servicetalk/examples/http/redirects/ManualRedirectClient.java

```diff
@@ -0,0 +1,122 @@
+
+                // Redirect of a GET request with a custom header:
+                HttpRequest originalGet = client.get("http://localhost:8080/sayHello")
+                        .addHeader(CUSTOM_HEADER, "value");
+                client.request(originalGet)
+                        .flatMap(response -> {
+                            if (response.status().statusClass() == REDIRECTION_3XX) {
+                                CharSequence location = response.headers().get(LOCATION);
+                                HttpClient redirectClient = lookupClient(location, client, secureClient);
+                                return redirectClient.request(redirectClient
+                                        .newRequest(originalGet.method(), location.toString())
+                                        .addHeader(CUSTOM_HEADER, originalGet.headers().get(CUSTOM_HEADER)));
+                            }
+                            // Decided not to follow redirect, return the original response or an error:
+                            return succeeded(response);
+                        })
+                        .afterFinally(responseProcessedLatch::countDown)
+                        .subscribe(resp -> {
+                            System.out.println(resp.toString((name, value) -> value));
+                            System.out.println(resp.payloadBody(textDeserializer()));
+                            System.out.println();
+                        });
+
+                responseProcessedLatch.await();
+
+                // Redirect of a POST request with a payload body:
+                responseProcessedLatch = new CountDownLatch(1);
+                HttpRequest originalPost = client.post("http://localhost:8080/sayHello")
+                        .payloadBody(client.executionContext().bufferAllocator().fromAscii("some_content"));
+                client.request(originalPost)
+                        .flatMap(response -> {
+                            if (response.status().statusClass() == REDIRECTION_3XX) {
+                                CharSequence location = response.headers().get(LOCATION);
+                                HttpClient redirectClient = lookupClient(location, client, secureClient);
+                                return redirectClient.request(redirectClient
+                                        .newRequest(originalPost.method(), location.toString())
+                                        .payloadBody(originalPost.payloadBody()));
+                            }
+                            // Decided not to follow redirect, return the original response or an error:
+                            return succeeded(response);
+                        })
+                        .afterFinally(responseProcessedLatch::countDown)
+                        .subscribe(resp -> {
+                            System.out.println(resp.toString((name, value) -> value));
```

Comment:
Is there any way to indicate whether redirect was taken in response or is it lost?

Reply:
It prints all meta-data of the response, the status code will also be there. Users will see 200 instead of 301.
Because it's just an example, let's avoid adding testing functionality and keep it consistent with the hello-world example.